lighthouse-extension/app/src/lighthouse-ext-background.js

```diff
@@ -111,14 +112,29 @@ async function runLighthouseInLR(connection, url, flags, {lrDevice, categoryIDs,
     config.settings.onlyCategories = categoryIDs;
   }
 
-  const results = await lighthouse(url, flags, config, connection);
-  if (!results) return;
+  try {
+    const results = await lighthouse(url, flags, config, connection);
+    if (!results) return;
 
-  if (logAssets) {
-    await assetSaver.logAssets(results.artifacts, results.lhr.audits);
-  }
+    if (logAssets) {
+      await assetSaver.logAssets(results.artifacts, results.lhr.audits);
+    }
+    return results.report;
+  } catch (err) {
+    // If an error ruined the entire lighthouse run, attempt to return a meaningful error.
+    if (!(err instanceof LHError) || !err.lhrRuntimeError) {
+      throw err;
+    }
+
+    const runtimeError = {
+      code: err.code,
+      message: err.friendlyMessage ?
+          `${err.friendlyMessage} (${err.message})` :
+          err.message,
+    };
 
-  return results.report;
+    return JSON.stringify(runtimeError, null, 2);
```

Comment:
I was thinking we're returning a psuedo LHR so one more level on top of this:


```js
return JSON.stringify({runtimeError}, null, 2);
```

Reply:
yeah I missed taht.